streams/tests/test_core.py

```diff
@@ -494,3 +494,39 @@ def test_partition_str():
 def test_stream_name_str():
     source = Stream(name='this is not a stream')
     assert str(source) == '<this is not a stream; Stream>'
+
+
+def test_zip_product():
+    a = Stream()
+    b = Stream()
+    c = a.zip_product(b)
+    d = a.combine_latest(b, emit_on=a)
+
+    L = c.sink_to_list()
+    L2 = d.sink_to_list()
+
+    a.emit(1)
+    a.emit(2)
+    b.emit('a')
+    b.emit('b')
+    a.emit(3)
+
+    assert L == [(1, 'a'), (2, 'a'), (3, 'b')]
+    assert L2 != L
```

Comment:
please be more explicit in the assert, write out what is expected, will help save mental energy in the long run
I assume it's `L2 == [(2, 'a'), (3, 'b')]`